agent/src/main/java/com/criteo/hadoop/garmadon/agent/tracers/hadoop/hdfs/FileSystemTracer.java

```diff
@@ -169,12 +169,13 @@ public static Object intercept(
     public static class WriteTracer extends MethodTracer {
         @Override
         protected ElementMatcher<? super TypeDescription> typeMatcher() {
-            return nameStartsWith("org.apache.hadoop.hdfs.DistributedFileSystem");
+            return nameStartsWith("org.apache.hadoop.hdfs.DistributedFileSystem").or(nameStartsWith("org.apache.hadoop.fs.Hdfs"));
         }
 
         @Override
         protected ElementMatcher<? super MethodDescription> methodMatcher() {
-            return named("create").and(takesArguments(7)).and(takesArgument(0, pathTD));
+            return named("create").and(takesArguments(7)).and(takesArgument(0, pathTD))
+                .or(named("createInternal"));
```

Comment:
no match on number of parameters?

Reply:
only one createInternal method exist on Hdfs class so I don't put any other filtering here